config/common.yml

```diff
@@ -163,12 +187,27 @@ services:
         '@PrestaShop\Module\PrestashopCheckout\PayPal\PaymentToken\EventSubscriber\PaymentMethodTokenEventSubscriber'
       ]
 
+  ps_checkout.event.dispatcher.symfony2:
+    class: 'Symfony\Component\EventDispatcher\EventDispatcherInterface'
+    factory: [ '@PrestaShop\Module\PrestashopCheckout\Event\SymfonyEventDispatcherFactory', "create" ]
```

Comment:
Quick question as I don't know the details of this module, but why do you need a specific event dispatcher? What does it do that the regular global event dispatcher from Symfony can't do?

Maybe it's because those services are also used on the FO, in a legacy environment that doesn't include the Symfony event dispatcher?

Reply:
Another question however, what is the difference between `ps_checkout.event.dispatcher.symfony` and `ps_checkout.event.dispatcher.symfony2` I mean I see that they don't have the same subscribers but why? 

Shouldn't the initial `ps_checkout.event.dispatcher.symfony` be cleaned if not used anymore? or maybe simply modified to match this new need?

Reply:
I added the second event dispatcher because we have a circular dependency in one command handler. So the first event dispatcher calls CaptureOrderCommandHandler and the second event dispatcher is injected into the command handler